### PR TITLE
Seed more methods with AotCompilerTools.

### PR DESCRIPTION
**Pull request by UltraNamahage**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [ ] I have provided test coverage for my change (where applicable)

### Description

If you upcast a class that implements a virtual generic method to a superclass or interface, the AoT build may not work properly.
They must be seeded appropriately.
[Details are explained here.](https://github.com/SixLabors/ImageSharp/issues/946#issuecomment-792549374)

### CODE REVIEW

**Comment by JimBobSquarePants** [comment]
This all looks excellent @UltraNamahage thanks ever so much!

I'm happy to approve this so let me know if you want to add anything else before I merge.

**Review by JimBobSquarePants — APPROVED** [review]

**Comment by UltraNamahage** [comment]
@JimBobSquarePants I'm sorry. There was still something missing.

**Comment by JimBobSquarePants** [comment]
Thanks! I wasn't aware of the `PreserveAttribute` functionality. Very useful!

**Pull request merged**
